libs/wallet/src/use-is-wallet-service-running.tsx

```diff
@@ -0,0 +1,42 @@
+import type { VegaConnector } from './connectors';
+import { useCallback, useEffect, useState } from 'react';
+import type { JsonRpcConnector } from './connectors';
+import { ClientErrors } from './connectors';
+
+export const useIsWalletServiceRunning = (
+  url: string,
+  connectors: { [key: string]: VegaConnector },
+  appChainId: string
+) => {
+  const [run, setRun] = useState<boolean | null>(null);
+
+  const checkState = useCallback(async () => {
+    const connector = connectors['jsonRpc'] as JsonRpcConnector;
+    connector.url = url;
+    try {
+      await connector.checkCompat();
+      const chainIdResult = await connector.getChainId();
+      if (chainIdResult.chainID !== appChainId) {
+        throw ClientErrors.WRONG_NETWORK;
+      }
+    } catch (e) {
+      return false;
+    }
+    return true;
+  }, [connectors, url, appChainId]);
+
+  useEffect(() => {
```

Comment:
(non blocking): I would probably change this slightly so that the checkState function handles state, and the useEffect handles the side effects only (the interval). Something like:

```
const [isRunning, setIsRunning] = useState<boolean | null>(null);

  const checkState = useCallback(async () => {
    const connector = connectors['jsonRpc'] as JsonRpcConnector;
    connector.url = url;

    try {
      await connector.checkCompat();
      const chainIdResult = await connector.getChainId();
      if (chainIdResult.chainID !== appChainId) {
        throw ClientErrors.WRONG_NETWORK;
      }
    } catch (e) {
      setIsRunning(false);
    }

    setIsRunning(true);
  }, [connectors, url, appChainId]);

  useEffect(() => {
    let interval: NodeJS.Timeout;

    checkState().then(() => {
      interval = setInterval(checkState, 1000 * 10);
    });

    return () => {
      clearInterval(interval);
    };
  }, [checkState]);
```

Reply:
yeah, it looks clearer little bit. Maybe in next iteration :-)